libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -704,13 +704,13 @@ bool ASTJsonConverter::visit(ElementaryTypeNameExpression const& _node)
 
 bool ASTJsonConverter::visit(Literal const& _node)
 {
-	char const* tokenString = Token::toString(_node.token());
+	string tokenString = tokenKind(_node.token());
 	Json::Value value{_node.value()};
 	if (!dev::validateUTF8(_node.value()))
 		value = Json::nullValue;
 	Token::Value subdenomination = Token::Value(_node.subDenomination());
 	std::vector<pair<string, Json::Value>> attributes = {
-		make_pair("token", tokenString ? tokenString : Json::Value()),
+		make_pair("kind", tokenString),
```

Comment:
I could do 'm_legacy ? "token" : "kind", tokenString' to have some degree of BC..
yet the old version of tokenString ('= Token::toString(_node.token()'); was never different from 'null', so in order to make it this field backwards compatible, we'd need to maintain a bug.